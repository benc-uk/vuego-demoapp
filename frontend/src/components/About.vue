<!--
About page
-----------------
Ben C, April 2018, Updated for Vue3 2021
-->

<template>
  <div class="card border-light mb-3 rounded-3">
    <h4 class="card-header bg-info"><i class="fas fa-info-circle fa-fw" /> About</h4>
    <div class="card-body">
      <div>Developed by Ben Coleman, 2018~2021</div>

      <ul class="list-group m-4">
        <li class="list-group-item d-flex justify-content-between align-items-center">
          App Version
          <span class="badge bg-primary rounded-pill p-2">{{ version }}</span>
        </li>
        <li class="list-group-item d-flex justify-content-between align-items-center">
          API Endpoint
          <span class="badge bg-primary rounded-pill p-2"> {{ apiEndpoint }}</span>
        </li>
      </ul>
    </div>
  </div>
</template>

<script>
const packageJson = require('../../package.json')

export default {
  data: function () {
    return {
      version: packageJson.version,
      apiEndpoint: process.env.VUE_APP_API_ENDPOINT || '/api'
    }
  }
}
</script>

<style scoped></style>
