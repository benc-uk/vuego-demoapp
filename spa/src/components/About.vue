<!--
About page
-----------------
Ben C, April 2018
-->

<template>
  <div class="card">
    <div class="card-header bg-success h3"><fa icon="info-circle" />&nbsp; About</div>
    <div class="card-body">
      Developed by Ben Coleman, 2018~2021<br /><br />
      <b>App Version:</b> {{ version }}<br />
      <b>API Endpoint:</b> {{ apiEndpoint }}
    </div>
  </div>
</template>

<script>
const packageJson = require('../../package.json')

export default {
  data: function () {
    return {
      version: packageJson.version,
      apiEndpoint: process.env.VUE_APP_API_ENDPOINT || '/api'
    }
  }
}
</script>
