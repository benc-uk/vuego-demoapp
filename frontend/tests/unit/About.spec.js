import { mount } from '@vue/test-utils'
import About from '@/components/About.vue'

describe('About.vue', () => {
  it('renders about screen', async () => {
    const wrapper = mount(About, {})

    expect(wrapper.html()).toMatchSnapshot()
  })
})
