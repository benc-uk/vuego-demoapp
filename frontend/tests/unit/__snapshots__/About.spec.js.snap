// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About.vue renders about screen 1`] = `
<div class="card border-light mb-3 rounded-3">
  <h4 class="card-header bg-info"><i class="fas fa-info-circle fa-fw"></i> About</h4>
  <div class="card-body">
    <div>Developed by Ben Coleman, 2018~2021</div>
    <ul class="list-group m-4">
      <li class="list-group-item d-flex justify-content-between align-items-center"> App Version <span class="badge bg-primary rounded-pill p-2">3.0.0</span></li>
      <li class="list-group-item d-flex justify-content-between align-items-center"> API Endpoint <span class="badge bg-primary rounded-pill p-2">/api</span></li>
    </ul>
  </div>
</div>
`;
